scripts/build_summary/index.html

```diff
@@ -164,28 +164,48 @@ <h1 v-else>Loading all of the JSON... please wait</h1>
           var r = Object.values(this.builds).countBy("repo")
           return r
         },
+        histogram_end: function(){
+          // Defines the most recent end of date based histogram.
+          // Buckets will be measured in units of 24 hours preceeding this point.
+
+          // Previously histogram end was always the current time, but this meant
+          // histogram buckets were the previous 24 hours before now, which didn't line
+          // up with the date labels on the graphs. It also meant that the values for
+          // each date varied during the day as a build could age enough to move up to
+          // the next bucket. This was exacerbated by a rounding error that caused
+          // builds for one day to be evenly split over two buckets
+          // (the correct day and the next day)
+          // Now histogram end is aligned with a day boundary so that the 24 hour
+          // buckets are aligned with calendar days.
+          end = new Date() // now
+          end.setHours(0,0,0,0) //start of today
+          end.setDate(end.getDate()+1) //start of tomorrow
+          return end
+        },
         labels: function(){
-          now = new Date()
           return [...Array(this.retention_days).keys()].map(
             (i) => {
               var daysAgo = this.retention_days-i
               var dayInMillis = 86400000
-              return new Date(now-dayInMillis * daysAgo).toLocaleDateString()
+              return new Date(this.histogram_end-dayInMillis * daysAgo).toLocaleDateString()
```

Comment:
nit: It'd be nice to see spaces around the operators here. I got confused initially, thinking that ``this.histogram_end-dayInMillis`` was a variable name.